### PR TITLE
Refactor FxOS Add-on statuses constants (bug 1207115)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1207115

### CODE REVIEW

File: docs/api/topics/firefox_os_addons.rst [hunk 1]
```diff
@@ -74,7 +74,7 @@ Detail
     :resjson string slug: The add-on slug (unique string identifier that can be used
         instead of the id to retrieve an add-on).
     :resjson string status: The add-on current status.
-        Can be "incomplete", "pending", or "public".
+        Can be "incomplete", *pending*, or *public*.
```
Comment (eviljeff) [inline docs/api/topics/firefox_os_addons.rst]: nit: "incomplete"

File: mkt/constants/base.py [hunk 2]
```diff
@@ -92,6 +93,18 @@
     'unlisted': STATUS_UNLISTED,
 }
 
+STATUS_FILE_CHOICES_API_v2 = {
+    STATUS_NULL: 'incomplete',
+    STATUS_PENDING: 'pending',
+    STATUS_PUBLIC: 'public',
+    STATUS_DISABLED: 'obsolete',
```
Comment (eviljeff) [inline mkt/constants/base.py]: STATUS_OBSOLETE?

File: mkt/constants/base.py [hunk 3]
```diff
@@ -92,6 +93,18 @@
     'unlisted': STATUS_UNLISTED,
 }
 
+STATUS_FILE_CHOICES_API_v2 = {
```
Comment (eviljeff) [inline mkt/constants/base.py]: Could we limit this to the valid choices we know are possible for add-ons (and apps)?

Comment (diox) [inline mkt/constants/base.py]: `STATUS_FILE_CHOICES` had the whole list so I also used the whole list here, even though some might be useless. I prefer to not limit this right now, I don't really want to dive deep into the code to find out which ones are not useful right now.

**Comment by eviljeff** [comment]
r+wc

**Pull request merged**